src/Marten/Events/Daemon/HighWater/HighWaterDetector.cs

```diff
@@ -119,7 +120,21 @@ private async Task<long> findCurrentMark(HighWaterStatistics statistics, Cancell
     {
         // look for the current mark
         _gapDetector.Start = statistics.SafeStartMark;
-        var current = await _runner.Query(_gapDetector, token).ConfigureAwait(false);
+        long? current;
+        try
+        {
+            current = await _runner.Query(_gapDetector, token).ConfigureAwait(false);
+        }
+        catch (InvalidOperationException e)
+        {
+            if (e.Message.Contains("An open data reader exists for this command"))
```

Comment:
**QUESTION:** Shouldn't that be a part of retry policy?

Reply:
I think this is pretty specific to this code, and if we put it in the retry policy, then someone customized the resiliency pipeline, you'd lose the specific behavior. Putting Polly here might not be a bad idea, but I wouldn't make that a high priority